src/services/activityService.ts

```diff
@@ -0,0 +1,105 @@
+import { ActivityOptions, ActivityType, Client } from "discord.js";
+import { selectRandom } from "../utils";
+
+type ActivityChoices = {
+    activity: ActivityType;
+    options: Array<string>;
+};
+
+const CHOICES: Array<ActivityChoices> = [
+    {
+        activity: "WATCHING",
+        options: [
+            "you... FROM INSIDE THE HOUSE",
+            "Jan Quadrant Vincent 16",
+            "Adventures of Baby Legs and Regular Legs",
+            "Avatar 86: This Time The Trees Are Real Mad",
+            "Seinfeld with Abyssal dub"
+        ]
+    },
+    {
+        activity: "PLAYING",
+        options: [
+            "Starcraft 6",
+            "Starfox 65",
+            "Phasmophob - OH SHIT HIDE",
+            "Bird Law Simulator 2037",
+            "Call Of Dookie",
+            "Sturgeon Simulator",
+            "Untitled Spruce Game",
+            "Pumpkin Carving Simulator 2020"
+        ]
+    },
+    {
+        activity: "LISTENING",
+        options: [
+            "Nefertiti's Fjord",
+            "We Didn't Start The Fire, but at 50% speed",
+            "John Malkovich gurgling marbles",
+            "Spooky Scary Skeletons"
+        ]
+    },
+    {
+        activity: "STREAMING",
+        options: ["a literal stream", "ice cubes melting in the sun", "my neighbor napping in a hammock"]
+    }
+];
+
+export class ActivityService {
+    private currentActivity: ActivityOptions = null;
+
+    constructor() {
+        this.currentActivity = this.newActivity();
+    }
+
+    initializeActivityTimeout(client: Client): void {
+        setTimeout(async () => {
```

Comment:
I made a simpler way to do timeouts if you want to try it out. (You can stick with this if you want though, nbd)
https://github.com/Awkewainze/FriendBot/blob/main/src/utils/timer.ts
https://github.com/Awkewainze/FriendBot/blob/main/src/utils/duration.ts

So to replace this, you could do
```typescript
private static running: boolean = false;
async function initializeActivityRandomization(client: Client): Promise<void> {
   // To avoid running multiple times.
    if (ActivityService.running) return;

    ActivityService.running = true;
    while (ActivityService.running) {
        await client.user.setActivity(this.newActivity());
        await Timer.for(Duration.between(Duration.fromMinutes(15), Duration.fromMinutes(60)))
            .start()
            .asAwaitable();
    }
}
```

Reply:
I'd prefer not to have static properties if possible and rely on the singleton instance to manage state. The rest of this looks good though! Will implement.